### PR TITLE
tidb-binlog: add configuration file

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

Add TiDB Binlog configuration file.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v4.0 changes for now)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/2133
- Other reference link(s):

### CODE REVIEW

File: reference/tidb-binlog/configs.md [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+* Default value: `true`
+
+#### kv_chan_cap
+
+* Specifies the number of write requests that the buffer can store before Pump receives these requests.
+* Default value: `1048576` (that is, 2 to the power of 20)
+
+#### slow_write_threshold
+
+* The threshold (in seconds) If it takes longer to write a single binlog file than this specified threshold, the write is considered slow write and `"take a long time to write binlog"` is output in the log.
```
Comment (WangXiangUSTC) [inline reference/tidb-binlog/configs.md]:
```suggestion
* The threshold (in seconds) if it takes longer to write a single binlog file than this specified threshold, the write is considered slow write and `"take a long time to write binlog"` is output in the log.
```

File: reference/tidb-binlog/configs.md [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+
+### pd-urls
+
+* The comma-separated list of PD URLs. If multiple addresses are specified, the PD client will automatically attempt to connect to another address if an error occurs when connecting to one address.
+* Default value: `"http://127.0.0.1:2379"`
+
+### initial-commit-ts
+
+* Specifies from which commit timestamp the replication task starts. This configuration is only applicable to the Drainer node that started replication for the first time. If a checkpoint already exists downstream, the replication will be performed according to the time recorded in the checkpoint.
+* Default value: `-1`。Drainer 会从 PD 得到一个最新的 timestamp 作为初始时间。Drainer will get a new timestamp from PD as the starting time.
```
Comment (WangXiangUSTC) [inline reference/tidb-binlog/configs.md]: have Chinese

**Comment by WangXiangUSTC** [comment]
LGTM

File: reference/tidb-binlog/configs.md [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+* Default value: `true`
+
+#### kv_chan_cap
+
+* Specifies the number of write requests that the buffer can store before Pump receives these requests.
+* Default value: `1048576` (that is, 2 to the power of 20)
+
+#### slow_write_threshold
+
+* The threshold (in seconds) if it takes longer to write a single binlog file than this specified threshold, the write is considered slow write and `"take a long time to write binlog"` is output in the log.
```
Comment (ran-huang) [inline reference/tidb-binlog/configs.md:104]:
```suggestion
* The threshold (in seconds). If it takes longer to write a single binlog file than this specified threshold, the write is considered slow write and `"take a long time to write binlog"` is output in the log.
```

File: reference/tidb-binlog/configs.md [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+* block-size
+* compaction-L0-trigger
+* compaction-table-size
+* compaction-total-size
+* compaction-total-size-multiplier
+* write-buffer
+* write-L0-pause-trigger
+* write-L0-slowdown-trigger
+
+For the detailed description of above items, see [GoLevelDB Document](https://godoc.org/github.com/syndtr/goleveldb/leveldb/opt#Options).
```
Comment (ran-huang) [inline reference/tidb-binlog/configs.md:127]:
```suggestion
For the detailed description of the above items, see [GoLevelDB Document](https://godoc.org/github.com/syndtr/goleveldb/leveldb/opt#Options).
```

File: reference/tidb-binlog/configs.md [hunk 5]
```diff
@@ -0,0 +1,351 @@
...
+
+## Drainer
+
+This section introduces the configuration items of Drainer. For the example of a complete Drainer configuration file, see [Drainer Configuration](https://github.com/pingcap/tidb-binlog/blob/master/cmd/drainer/drainer.toml)
+
+### addr
+
+* Specifies the listening address of HTTP API in the format of `host: port`.
```
Comment (ran-huang) [inline reference/tidb-binlog/configs.md:135]:
```suggestion
* Specifies the listening address of HTTP API in the format of `host:port`.
```

File: reference/tidb-binlog/configs.md [hunk 6]
```diff
@@ -0,0 +1,351 @@
...
+* Default value: `"127.0.0.1:8249"`
+
+### advertise-addr
+
+* Specifies the externally accessible HTTP API address. This address is registered in PD in the format of `host: port`.
```
Comment (ran-huang) [inline reference/tidb-binlog/configs.md:140]:
```suggestion
* Specifies the externally accessible HTTP API address. This address is registered in PD in the format of `host:port`.
```

File: reference/tidb-binlog/configs.md [hunk 7]
```diff
@@ -0,0 +1,351 @@
...
+* Default value: `hostname:port number`. For example, `node-1:8249`.
+
+### data-dir
+
+* Specifies the directory used to store files that need to be saved during Drainer operation.
+* Default value: `"data.drainer"`
+
+### detect-interval
+
+* Specifies the interval (in seconds) at which PD updates the Pump information
```
Comment (ran-huang) [inline reference/tidb-binlog/configs.md:165]:
```suggestion
* Specifies the interval (in seconds) at which PD updates the Pump information.
```

File: reference/tidb-binlog/configs.md [hunk 8]
```diff
@@ -0,0 +1,351 @@
...
+* Default value: `5`
+
+### pd-urls
+
+* The comma-separated list of PD URLs. If multiple addresses are specified, the PD client will automatically attempt to connect to another address if an error occurs when connecting to one address.
+* Default value: `"http://127.0.0.1:2379"`
+
+### initial-commit-ts
+
+* Specifies from which commit timestamp the replication task starts. This configuration is only applicable to the Drainer node that started replication for the first time. If a checkpoint already exists downstream, the replication will be performed according to the time recorded in the checkpoint.
```
Comment (ran-huang) [inline reference/tidb-binlog/configs.md:175]:
```suggestion
* Specifies from which commit timestamp the replication task starts. This configuration is only applicable to the Drainer node that starts replication for the first time. If a checkpoint already exists downstream, the replication will be performed according to the time recorded in the checkpoint.
```

File: reference/tidb-binlog/configs.md [hunk 9]
```diff
@@ -0,0 +1,351 @@
...
+* Default value: `-1`. Drainer will get a new timestamp from PD as the starting time.
+
+### synced-check-time
+
+* You can access the `/ status` path through the HTTP API to query the status of Drainer replication. `synced-check-time` specifies how many minutes from the last successful replication is considered as `synced`, that is, the replication is complete.
```
Comment (ran-huang) [inline reference/tidb-binlog/configs.md:180]:
```suggestion
* You can access the `/status` path through the HTTP API to query the status of Drainer replication. `synced-check-time` specifies how many minutes from the last successful replication is considered as `synced`, that is, the replication is complete.
```

File: reference/tidb-binlog/configs.md [hunk 10]
```diff
@@ -0,0 +1,351 @@
...
+Default value: `[]`
+
+#### txn-batch
+
+* When the downstream is the `mysql` or `tidb` type, DML operations are executed in different batches. This parameter specifies how many DML operations can be included in each transaction.
+* Default value: `20`
+
+#### worker-count
+
+* When the downstream is the `mysql` or `tidb` type, DML operations are executed in different batches. This parameter specifies the concurrency numbers of DML operations.
```
Comment (ran-huang) [inline reference/tidb-binlog/configs.md:287]:
```suggestion
* When the downstream is the `mysql` or `tidb` type, DML operations are executed concurrently. This parameter specifies the concurrency numbers of DML operations.
```

File: reference/tidb-binlog/configs.md [hunk 11]
```diff
@@ -0,0 +1,351 @@
...
+* Default value: `16`
+
+#### disable-dispatch
+
+* Disables the concurrency and forcibly set `work-count` to `1`.
```
Comment (ran-huang) [inline reference/tidb-binlog/configs.md:292]:
```suggestion
* Disables the concurrency and forcibly set `worker-count` to `1`.
```

File: reference/tidb-binlog/configs.md [hunk 12]
```diff
@@ -0,0 +1,351 @@
...
+* Default value: `false`
+
+#### safe-mode
+
+If Safe mode is enabled, Drainer modifies the replication updates in the following way:
```
Comment (ran-huang) [inline reference/tidb-binlog/configs.md:297]:
```suggestion
If the safe mode is enabled, Drainer modifies the replication updates in the following way:
```

File: reference/tidb-binlog/configs.md [hunk 13]
```diff
@@ -0,0 +1,351 @@
...
+This document introduces the configuration items of TiDB Binlog.
+
+## Pump
+
+This section introduces the configuration items of Pump. For the example of a complete Pump configuration file, see [Pump Configuration](https://github.com/pingcap/tidb-binlog/blob/master/cmd/pump/pump.toml).
+
+### addr
+
+* Specifies the listening address of HTTP API in the format of `host: port`.
+* Default value: `"127.0.0.1:8250"`
```
[the remark this answers is not in the document]
Comment (TomShawn) [reply reference/tidb-binlog/configs.md:17]: It is not valid. Removed in this PR, and I'll create another PR to modify the Chinese document.

**Comment by TomShawn** [comment]
@ran-huang PTAL again, thanks!

File: TOC.md [hunk 14]
```diff
@@ -327,6 +327,9 @@
     - [Deploy](/reference/tidb-binlog/deploy.md)
     - [Maintain](/reference/tidb-binlog/maintain.md)
     - [Monitor](/reference/tidb-binlog/monitor.md)
+    + [Configure](/reference/tidb-binlog/config.md)
+      - [Pump](/reference/tidb-binlog/configs.md#pump)
+      - [Drainer](/reference/tidb-binlog/configs.md#drainer)
```
Comment (ran-huang) [inline TOC.md:332]:
```suggestion
      - [Pump](/reference/tidb-binlog/config.md#pump)
      - [Drainer](/reference/tidb-binlog/config.md#drainer)
```

**Pull request merged**